inmem_transport.go

```diff
@@ -287,6 +288,9 @@ func (i *inmemPipeline) decodeResponses() {
 			case <-i.shutdownCh:
 				return
 			}
+
+			timer.Stop()
```

Comment:
This should be deferred on L267 in case of an early return in the `select` block above

Reply:
@kisunji if we put defer in L276, it will be defer in loop and absolutely bad practice. How do you think?

Reply:
Ah, that's a good catch. We may have to call `timer.Stop` in all of the branches that `return` early then

Reply:
It depends on each logic. As far as I look at the source code. We just need to invoke stop at the end of select or just inside a branch.